org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/servicetemplates/selfserviceportal/SelfServicePortalResource.java

```diff
@@ -126,7 +142,14 @@ public Response putImage(@FormDataParam("file") InputStream uploadedInputStream,
             throw new WebApplicationException(e);
         }
         RepositoryFileReference ref = new RepositoryFileReference(this.id, "image.jpg");
-        return RestUtils.putContentToFile(ref, uploadedInputStream, body.getMediaType());
+        Response response = RestUtils.putContentToFile(ref, uploadedInputStream, body.getMediaType());
```

Comment:
I think you should do this in a separate method and also perform these steps when other files are uploaded also.